test/new-e2e/tests/agent-metric-logs/log-agent/utils/file_tailing_utils.go

```diff
@@ -3,28 +3,45 @@
 // This product includes software developed at Datadog (https://www.datadoghq.com/).
 // Copyright 2016-present Datadog, Inc.
 
-package logagent
+// utils provides shared common functions so different E2E tests suites can use them.
+package utils
 
 import (
 	"encoding/json"
 	"fmt"
 	"strings"
+	"testing"
 	"time"
 
+	"github.com/stretchr/testify/assert"
+
 	"github.com/DataDog/datadog-agent/test/fakeintake/aggregator"
 	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+
 	componentos "github.com/DataDog/test-infra-definitions/components/os"
-	"github.com/stretchr/testify/assert"
 )
 
-// appendLog appen log with 'content', which is then repeated 'reccurrence' times and verifies log contents.
-func appendLog(s *LinuxFakeintakeSuite, content string, recurrence int) {
+type TestHelper interface {
+	T() *testing.T
+	Env() *e2e.FakeIntakeEnv
+	AppendLog(content string, recurrence int)
+	CheckLogFilePresence(logPath string)
+	CheckLogs(service, content string, expectLogs bool)
+	CleanUp()
+	EventuallyWithT(assertFunction func(*assert.CollectT), waitFor, tick time.Duration, option ...interface{}) bool
+	IsDevMode() bool
```

Comment:
<!-- Warning highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes and they are blockers. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question. -->
**🔨 warning**
Having this in the interface sounds dangerous: you could have an infinite loop if you were calling `th.AppendLog` from `AppendLog`.

What you need here is the testing context, the fakeintake and the VM: I would rather explicitly pass those 3 as params, or have an interface with only those 3, no actions !
```suggestion
	IsDevMode() bool
```